server/lib/transactions.js

```diff
@@ -34,23 +35,29 @@ export function createFromPaidExpense(paymentMethod, expense, paymentResponses,
 
   // We assume that all expenses are in Group currency
   // (otherwise, ledger breaks with a triple currency conversion)
-
-  return models.Transaction.create({
+  const transaction = {
     netAmountInGroupCurrency: -1*(expense.amount + fees),
-    amountInTxnCurrency: -expense.amount,
-    paymentProcessorFeeInTxnCurrency: fees,
     txnCurrency,
-    txnCurrencyFxRate: 1,
+    paymentProcessorFeeInTxnCurrency: fees,
     ExpenseId: expense.id,
     type: type.EXPENSE,
     amount: -expense.amount,
     currency: expense.currency,
     description: expense.title,
     UserId,
     GroupId: expense.GroupId,
-  })
-  .tap(t => paymentMethod ? t.setPaymentMethod(paymentMethod) : null)
-  .then(t => createPaidExpenseActivity(t, paymentResponses, preapprovalDetails));
+    HostId: host.id
+  };
+
+  return getFxRate(expense.currency, txnCurrency, expense.incurredAt || expense.createdAt)
```

Comment:
Yep. Fixed. Still using this for add funds or paid manually (if expense.currency != host.currency)